Objects/unicodeobject.c

```diff
@@ -14309,83 +14309,65 @@ unicode_getnewargs(PyObject *v, PyObject *Py_UNUSED(ignored))
 }
 
 /*
-This function searchs the longest common leading whitespace
-of all lines in the [src, end).
-It returns the length of the common leading whitespace and sets `output` to
-point to the beginning of the common leading whitespace if length > 0.
+   Find the longest common leading whitespace among a list of lines.
+   Whitespace-only lines are ignored.
+   Returns the margin length (>= 0).
 */
 static Py_ssize_t
-search_longest_common_leading_whitespace(
-    const char *const src,
-    const char *const end,
-    const char **output)
-{
-    // [_start, _start + _len)
-    // describes the current longest common leading whitespace
-    const char *_start = NULL;
-    Py_ssize_t _len = 0;
-
-    for (const char *iter = src; iter < end; ++iter) {
-        const char *line_start = iter;
-        const char *leading_whitespace_end = NULL;
-
-        // scan the whole line
-        while (iter < end && *iter != '\n') {
-            if (!leading_whitespace_end && *iter != ' ' && *iter != '\t') {
-                /* `iter` points to the first non-whitespace character
-                   in this line */
-                if (iter == line_start) {
-                    // some line has no indent, fast exit!
-                    return 0;
-                }
-                leading_whitespace_end = iter;
-            }
-            ++iter;
-        }
+search_longest_common_leading_whitespace(PyObject *lines, Py_ssize_t nlines)
```

Comment:
Please don't refactor this. It may be performance critical so we really don't want this to be slower. Make it use raw pointers.